### ISSUE TITLE
[Adaptive] Inputs alway re-prompt even when they have a value set externally

### ISSUE THREAD

**Issue by DingmaomaoBJTU** [microsoft/botbuilder-dotnet#2566]
## Version
What package version of the SDK are you using.

## Describe the bug
**[In one dialog]** When the text prompt is already prompt with asking attribute Dialog.A, and the flow has been interrupted into another flow which will set the attribute Dialog.A, then when the flow is back with this text prompt, it still reprompt even if the attribute A is already set.
[CreateMeeting.dialog.txt](https://github.com/microsoft/botbuilder-dotnet/files/3635032/CreateMeeting.dialog.txt)

The scenarios I want to support is:
Main flow: setAttendees -> setTitle -> setContent...
GetInput flow: Analysis entitles -> setAttendees
After switch back from GetInput flow, I hope the setAtendees text prompt won't reprompt as the property is already set.

## To Reproduce
Add a text prompt in main flow, binding with dialog.a
Add a intent trigger flow, in this flow, set dialog.a to some value.
Run the bot.
First, the bot will prompt the text prompt "sure, who will join"
Then trigger the intent with "add qiong", and dialog.a will be set with qiong
Then the flow will be back to main flow. "sure, who will join" prompt up again.

![image](https://user-images.githubusercontent.com/16643193/65296691-cdc07e80-db97-11e9-8cfb-c402cd232a0a.png)

## Expected behavior
The text prompt should not be reprompt.

## Additional context

[bug]

**Comment by vishwacsena** [comment]
I just verified that this does not repro with the latest bits in the SDK (4.Future branch). 

```json
{
    "$type": "Microsoft.AdaptiveDialog",
    "$schema": "../../../schemas/sdk.schema",
    "generator": "root.lg",
    "recognizer": {
        "$type": "Microsoft.RegexRecognizer",
        "intents": [
          {
    
              "$type": "Microsoft.IntentPattern",
              "intent": "NoName",
              "pattern": "(?i)no" 
          }
        ]
    },
    "triggers": [
        {
            "$type": "Microsoft.OnBeginDialog",
            "actions": [
                {
                    "$type": "Microsoft.TextInput",
                    "property": "$name",
                    "prompt": "What is your name?"
                },
                {
                    "$type": "Microsoft.SendActivity",
                    "activity": "I have {$name} as your name"
                }
            ]
        },
        {
            "$type": "Microsoft.OnIntent",
            "intent": "NoName",
            "actions": [
                {
                    "$type": "Microsoft.SetProperty",
                    "property": "$name",
                    "value": "'Human'"
                }
            ]
        }
    ]
}
```

Scenario 1: 
```markdown
user : hi
bot : what is your name?
user: vishwac
bot : I have vishwac as your name
```

Scenario 2: 
```markdown
user : hi
bot : what is your name?
user : no
bot : I have Human as your name
```

@DingmaomaoBJTU - it would be good to if you can help narrow down the repro. 

Closing as no repro for now.

**Comment by vishwacsena** [comment]
Reactivating. I have a repro. The issue appears to be related to SetProperty doing special logic to clear state so the next reprompt will re-evaluate new state. We need to do this in a more generic way and not focus on just setProperty.

**Comment by vishwacsena** [comment]
Need to chat with @DingmaomaoBJTU on the validity of the scenario she is modelling to assess the priority. Tagging as P1 for now.

**Comment by DingmaomaoBJTU** [comment]
[testInterruptionLUIS1.zip](https://github.com/microsoft/botbuilder-dotnet/files/3645069/testInterruptionLUIS1.zip)

For more generic scenario you can check this. What I want to do is use an intent handler to resolve luis result. Please correct me if you have better suggestion.

**Comment by vishwacsena** [comment]
Verify repro in latest 4.Future branch for interruption intent with sendActivity + set property.

**Comment by vishwacsena** [comment]
Verified I can repro this with the latest in 4.Future branch. 

```C#
var rootDialog = new AdaptiveDialog(nameof(AdaptiveDialog))
{
                
    Generator = new TemplateEngineLanguageGenerator(new TemplateEngine().AddFile(lgFile)),
    Recognizer = new RegexRecognizer()
    {
        Intents = new List<IntentPattern>() {
            new IntentPattern()
            {
                Intent = "NoName",
                Pattern = "(?i)no"
            }
        }
    },
    //AutoEndDialog = false,
    Triggers = new List<OnCondition>()
    {
        new OnBeginDialog()
        {
            Actions = new List<Dialog>()
            {
                new TextInput() {
                    Prompt = new ActivityTemplate("What is your name?"),
                    Property = "$userName"
                },
                new SendActivity("I have {$userName} as your name")
            }
        },
        new OnIntent()
        {
            Intent = "NoName",
            Actions = new List<Dialog>()
            {
                new SetProperty()
                {
                    Property = "$userName",
                    Value = "'Human'"
                }
// Commenting or uncommenting this line aka having additional actions after the set property causes different behavior with how inputs behave.
                ,new SendActivity("I'm going with {$userName}")
            }
        }
    }
};
```

**Expected**: 
```markdown
bot: what is your name?
user: no
bot: I'm going with Human
bot: I have Human as your name
```

**Actual**: 
```markdown
bot: What is your name?
user: no
bot: I'm going with Human
bot: What is your name?
```

**Comment by vishwacsena** [comment]
Verified this continues to repro in master.

**Comment by Stevenic** [comment]
Should be fixed by #2735
